internal/manager/run.go

```diff
@@ -189,15 +187,27 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 	}
 
 	if c.Konnect.ConfigSynchronizationEnabled {
+		// In case of failures when building Konnect related objects, we're not returning errors as Konnect is not
+		// considered critical feature, and it should not break the basic functionality of the controller.
+
 		setupLog.Info("Start Konnect client to register runtime instances to Konnect")
 		konnectClient, err := konnect.NewClient(c.Konnect)
 		if err != nil {
-			return fmt.Errorf("failed to create konnect client: %w", err)
+			setupLog.Error(err, "failed creating konnect client, skipping running NodeAgent")
+		} else {
+			hostname, _ := os.Hostname()
+			version := metadata.Release
+			agent := konnect.NewNodeAgent(hostname, version, setupLog, konnectClient)
+			agent.Run()
+		}
+
+		konnectAdminAPIClient, err := adminapi.NewKongClientForKonnectRuntimeGroup(ctx, c.Konnect)
+		if err != nil {
+			setupLog.Error(err, "failed creating Konnect Runtime Group Admin API client, skipping synchronisation")
+		} else {
+			setupLog.Info("Initialized Konnect Admin API client")
+			clientsManager.SetKonnectClient(konnectAdminAPIClient)
```

Comment:
If the above call to `konnect.NewClient()` fails, does it make sense to call this? 🤔

Reply:
Yes, I think so. The above `konnectClient` is in reality the Node API clien and it's not affecting Admin API functionality (as you can see it's not passed down to the `konnectAdminAPIClient`). I think it's worth trying our best when setting things up.

Reply:
What is the purpose of running this agent?

Do I understand it correctly that if the node agent gets started but then something fails when starting the Admin API client for konnect we'll register the node but e.g. the configuration won't be sent (and hence showed on konnect's UI)?

Reply:
Yup, that's all true. Node agent is meant for sending "ping" messages to Konnect so that KIC is available as a node in Runtime Group. It'll also register nodes for all gateways we manage (it's WIP by Tao).

Reply:
I renamed the client that's used for Node APIs so it's telling directly what we use it for in the current state. c841e60